src/metabase/pulse/render/body.clj

```diff
@@ -1030,16 +1030,31 @@
       [:img {:style (style/style {:display :block :width :100%})
              :src   (:image-src image-bundle)}]]}))
 
+(defn- all-unique?
+  [funnel-rows]
+  (let [ks (into #{} (map :key funnel-rows))]
+    (= (count ks) (count funnel-rows))))
+
+(defn- build-funnel-rows
+  [funnel-rows raw-rows]
+  (let [funnel-rows (map-indexed (fn [idx row] (assoc row :idx (inc idx))) funnel-rows)
+        xf-rows     (into {} (map (fn [row] (vec (take-last 2 row))) raw-rows))]
+    (for [{idx     :idx
+           k       :key
+           enabled :enabled} funnel-rows]
+      (when enabled
```

Comment:
The PR description doesn't mention this condition, and it wasn't present in the old code. Can you update the PR to mention and explain this change? 

Also, do you mean to leave `nil` values in this list when the row is not enabled? There's a very neat way to filter them out, which is to replace:

```clojure
(for [{..., enabled :enabled} funnel-rows]
  (when enabled
    ...))
```

with just

```clojure
(for [{..., enabled? :enabled} funnel-rows
      :when enabled?]
  ...)